providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdGrpcInterceptor.java

```diff
@@ -0,0 +1,56 @@
+package dev.openfeature.contrib.providers.flagd;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.ForwardingClientCall;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapSetter;
+
+import javax.annotation.Nullable;
+
+/**
+ * FlagdGrpcInterceptor is an interceptor for grpc communication from java-sdk to flagd.
+ * <a href="https://github.com/open-telemetry/opentelemetry-java-docs">credits</a>
+ */
+final class FlagdGrpcInterceptor implements ClientInterceptor {
+    private static final TextMapSetter<Metadata> SETTER = new Setter();
+
+    private final OpenTelemetry openTelemetry;
+
+    FlagdGrpcInterceptor(final OpenTelemetry openTelemetry) {
+        this.openTelemetry = openTelemetry;
+    }
+
+    @Override public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(MethodDescriptor<ReqT, RespT> methodDescriptor,
+                                                                         CallOptions callOptions, Channel channel) {
+
+        final ClientCall<ReqT, RespT> call = channel.newCall(methodDescriptor, callOptions);
+
+        return new ForwardingClientCall.SimpleForwardingClientCall<ReqT, RespT>(call) {
+            @Override
+            public void start(Listener<RespT> responseListener, Metadata headers) {
+                openTelemetry.getPropagators().getTextMapPropagator().inject(Context.current(), headers, SETTER);
```

Comment:
I never used Lombok, but I think this is not safe since https://github.com/open-feature/java-sdk-contrib/pull/294/files#diff-f368ee4f4edf9090cfaf37caa1e694ef41d8e5753ac86a3dc05f1381f133bc0dR85 doesn't have a default value